core/src/main/java/io/questdb/cutlass/http/client/HttpClient.java

```diff
@@ -419,26 +484,40 @@ private void connect(CharSequence host, int port) {
             nf.configureKeepAlive(fd);
             long addrInfo = nf.getAddrInfo(host, port);
             if (addrInfo == -1) {
-                disconnect();
+                nf.close(fd);
                 throw new HttpClientException("could not resolve host ").put("[host=").put(host).put("]");
             }
             if (nf.connectAddrInfo(fd, addrInfo) != 0) {
                 int errno = nf.errno();
-                disconnect();
+                nf.close(fd);
                 nf.freeAddrInfo(addrInfo);
                 throw new HttpClientException("could not connect to host ").put("[host=").put(host).put(", port=").put(port).put(", errno=").put(errno).put(']');
             }
             nf.freeAddrInfo(addrInfo);
+
+            if (nf.configureNonBlocking(fd) < 0) {
+                int errno = nf.errno();
+                nf.close(fd);
+                throw new HttpClientException("could not configure socket to be non-blocking [fd=").put(fd).put(", errno=").put(errno).put(']');
+            }
             socket.of(fd);
+            if (socket.supportsTls()) {
+                String serverName = Chars.toString(host);
```

Comment:
We already have `Chars.toString()` call in `JavaTlsClientSocket#startTlsSession()`, so this allocation is redundant.

Reply:
well spotted! it's a left-over from before.

Reply:
will do a separate PR